internal/authorization/admin.go

```diff
@@ -0,0 +1,36 @@
+package authorization
+
+import (
+	"context"
+	"fmt"
+)
+
+const ADMIN_OBJECT = "privileged:user"
+
+func (a *Authorizer) CreateAdmin(ctx context.Context, username string) error {
+	ctx, span := a.tracer.Start(ctx, "authorization.Authorizer.CreateAdmin")
+	defer span.End()
+
+	user := fmt.Sprintf("user:%s", username)
+	err := a.client.WriteTuple(ctx, user, "admin", ADMIN_OBJECT)
+	return err
+}
+
+func (a *Authorizer) RemoveAdmin(ctx context.Context, username string) error {
+	ctx, span := a.tracer.Start(ctx, "authorization.Authorizer.RemoveAdmin")
+	defer span.End()
+
+	user := fmt.Sprintf("user:%s", username)
+	err := a.client.DeleteTuple(ctx, user, "admin", ADMIN_OBJECT)
+	return err
+}
+
+func (a *Authorizer) CheckAdmin(ctx context.Context, username string) (bool, error) {
+	ctx, span := a.tracer.Start(ctx, "authorization.Authorizer.CheckAdmin")
+	defer span.End()
+
+	user := fmt.Sprintf("user:%s", username)
+	allowed, err := a.Check(ctx, user, "admin", ADMIN_OBJECT)
+
+	return allowed, err
+}
```

Comment:
this is a bit confusing, as `struct` is defined in a different file

you could be creating a separate struct for these functions and use [composition](https://www.codecademy.com/resources/docs/go/composition)

that would make it siloed

Reply:
makes sense, changed it.